services/motion/builtin/builtin.go

```diff
@@ -122,18 +126,31 @@ func (ms *builtIn) Reconfigure(
 	ms.movementSensors = movementSensors
 	ms.slamServices = slamServices
 	ms.components = components
+	if ms.state != nil {
+		ms.state.Stop()
+	}
+	ms.state = state.NewState(ms.logger)
 	return nil
 }
 
 type builtIn struct {
 	resource.Named
-	resource.TriviallyCloseable
 	fsService       framesystem.Service
 	movementSensors map[resource.Name]movementsensor.MovementSensor
 	slamServices    map[resource.Name]slam.Service
 	components      map[resource.Name]resource.Resource
 	logger          golog.Logger
-	lock            sync.Mutex
+	lock            sync.RWMutex
+	state           *state.State
+	closed          bool
```

Comment:
closed tracks whether or not the motion service has been closed. Protected by `lock`